### DISCUSSION TITLE
Use of `StackNavigation.subscribe`

### DISCUSSION THREAD

**Discussion by YektaDev** [arkivanov/Decompose#495] [Q&A] [answered]
Hi.

I was exploring the library once more🙂. I found something that I couldn't think of an answer for. Since both the stack navigation and the child stack value can both be observed, in a component containing both, what is the utility of observing `StackNavigation` when there's already the ability to observe `Value<ChildStack<*, *>>`? The latter provides way more information than what stack navigation's observer does. Isn't it true?

Thanks in advance!

**Comment by arkivanov** [comment, marked as answer]
`Child Stack` observes `StackNavigation`. Usually you don't need to observe it in your code.

Here is some context. A while ago there was no `StackNavigation`, the `childStack` function was returning `StackRouter` interface, which contained both the observable stack and the `navigate` function. This caused "Chicken and egg" problems, when you create a child component in the `childFactory` function, and you want to access `StackRouter` for navigation, which is not yet available. This was fixed by extracting the navigation part to the separate `StackNavigation`, which is created beforehand.
